pkg/asset/installconfig/gcp/client.go

```diff
@@ -222,6 +225,34 @@ func (c *Client) GetProjects(ctx context.Context) (map[string]string, error) {
 	return projects, nil
 }
 
+// GetRegions gets the regions that are valid for the project. An error is returned when unsuccessful
+func (c *Client) GetRegions(ctx context.Context, project string) (map[string]string, error) {
+	computeCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	svc, err := c.getComputeService(computeCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	regionCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+	gcpRegionsList, err := svc.Regions.List(project).Context(regionCtx).Do()
```

Comment:
I am fine with merging this PR using the existing pattern of sharing contexts, although I disagree with it for the reasons mentioned above. Here is an example of the existing pattern: https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/gcp/client.go#L72-L85

If we go that route, I would prefer this be updated to use the same variable naming conventions in case we decide to fix the problem later in bulk.

I'm also happy to have the pattern fixed here, in a separate commit.

Reply:
This is simple enough to add to this PR in a separate commit. If I follow the pattern the following functions will be changed to match the pattern:

```
GetMachineType
GetNetwork
GetSubnetworks
GetZones
```

The `getComputeService` will be altered to make the child context.